### ISSUE TITLE
Fix panic issue observed in setSiteImplicitly function

### ISSUE THREAD

**Issue by ashishshinde-pubm** [prebid/prebid-server#3052]
Prebid has **setSitePageIfEmpty** function which is getting called from **setSiteImplicitly** in endpoints/openrtb2/auction.go
This function will never set the value of site object present under RequestWrapper. 

```
func setSitePageIfEmpty(site *openrtb2.Site, sitePage string) {
	if site == nil {
		site = &openrtb2.Site{}
	}
	if site.Page == "" {
		site.Page = sitePage
	}
}
```

The call is being made as below 
```
func setSiteImplicitly(httpReq *http.Request, r *openrtb_ext.RequestWrapper) {
	/// some code
	if referrerCandidate != "" {
		setSitePageIfEmpty(r.Site, referrerCandidate) // this will never set the value of r.Site if its nil object
		if parsedUrl, err := url.Parse(referrerCandidate); err == nil {
			setSiteDomainIfEmpty(r.Site, parsedUrl.Host) // this function panics since it tries to set the value of r.Site.Domain
		}
	}
}
```

This PR handles the panic case, please review.
https://github.com/prebid/prebid-server/pull/3053

**Comment by hhhjort** [comment]
PR merged

**Comment by SyntaxNode** [comment]
Released with PBS-Go 0.273.0.
